app/Console/Commands/MakeFilamentActionCommand.php

```diff
@@ -0,0 +1,169 @@
+                            {--custom-component : Generate a custom component action}
+                            {--infolist : Generate an infolist component action}
+                            {--notification : Generate a notification action}
+                            {--global-search : Generate a global search result action}';
+
+    protected $description = 'Create a new Filament action class';
+
+    /**
+     * @var array<string, string>
+     */
+    protected array $actionTypes = [
+        'form' => 'Filament\\Forms\\Components\\Actions\\Action',
+        'table' => 'Filament\\Tables\\Actions\\Action',
+        'table-bulk' => 'Filament\\Tables\\Actions\\BulkAction',
+        'custom-component' => 'Filament\\Actions\\Action',
+        'infolist' => 'Filament\\Infolists\\Components\\Actions\\Action',
+        'notification' => 'Filament\\Notifications\\Actions\\Action',
+        'global-search' => 'Filament\\GlobalSearch\\Actions\\Action',
+    ];
+
+    /**
+     * @throws FileNotFoundException
+     */
+    public function handle(): void
+    {
+        $name = $this->getNameArgument();
+        $className = $this->getClassName($name);
+        $type = $this->getActionType();
+        $actionClass = $this->actionTypes[$type];
+        $path = $this->getFilePath($className, $type);
+
+        if ($this->fileExists($path)) {
+            warning("$className already exists.");
+
+            return;
+        }
+
+        $stubContent = $this->getStubContent();
+        $this->generateActionFile($path, $className, $actionClass, $stubContent);
+
+        info("$className created successfully at:");
+        info($path);
+    }
+
+    private function getNameArgument(): string
+    {
+        return $this->argument('name') ?? text(
+            label: 'Enter the name of the Filament Action',
+            placeholder: 'MyActionName',
+            required: 'Action name is required.',
+            hint: "We will append 'Action' to this name, so no need to include it."
+        );
+    }
+
+    private function getClassName(string $name): string
+    {
+        return Str::studly($name).'Action';
+    }
+
+    private function getActionType(): string
+    {
+        $selectedTypes = array_filter([
+            'form' => $this->option('form'),
+            'table' => $this->option('table'),
+            'table-bulk' => $this->option('table-bulk'),
+            'custom-component' => $this->option('custom-component'),
+            'infolist' => $this->option('infolist'),
+            'notification' => $this->option('notification'),
+            'global-search' => $this->option('global-search'),
+        ]);
+
+        if (count($selectedTypes) > 1) {
+            warning('Please specify only one action type.');
+            exit;
+        }
+
+        if (count($selectedTypes) === 0) {
+            $selectedType = select(
+                label: 'What type of action is this for?',
+                options: [
```

Comment:
Since we are talking about actions i dont think we needed to include the word "action" in the options.

- Form
- Table
- TableBulk
- Infolist
- Notification
- Global Search (Im not sure what this is TBH)
- Custom

Reply:
@CodeWithDennis Not a use case I can see that will get utilised often, but it's supported by Filament so I included it: https://filamentphp.com/docs/3.x/actions/overview#global-search-result-actions